Engine/entry1.c

```diff
@@ -134,6 +134,9 @@ OENTRY opcodlst_1[] = {
   { "midglobal",S(MIDGLOBAL),0,   "",     "Sm", midglobal, NULL, NULL, NULL},
   { "ihold",  S(LINK),0,          "",     "",     ihold, NULL, NULL, NULL  },
   { "turnoff",S(LINK),0,           "",     "",     NULL,   turnoff, NULL, NULL },
+  { "offsetsmps", S(AOP), 0, "i", "", (SUBR) sa_offset, NULL, NULL},
+  { "offsetsmps", S(AOP), 0, "k", "", NULL,(SUBR) sa_offset, NULL},
+  { "earlysmps", S(AOP), 0, "k", "", NULL,(SUBR) sa_early, NULL},
```

Comment:
Only question I have for this PR is an earlysmps should be added with i-rate output, to match the offsetsmps with i-rate output. I suppose it's possible someone might want to do init-time processing and the event both starts mid-buffer and ends before the end of the ksmps buffer (if there's a large ksmps size).

Reply:
the problem is that earlysmps will only return 0 at i-time. It is set at the last cycle of performance.

Reply:
Confused then here, so if an event both starts mid-buffer and ends before end of buffer (e.g., ksmps = 512, event starts 32 smps in, ends after 64 smps), there's no way to know earlysmps at init-time (in native code too)?  Is this a deeper problem that needs further internal work?

Reply:
What I am saying is that the  variable is set at the last cycle, before that it's zero. That's how it works. It does not make sense to have it set at init time (it's only meaningful at perf time anyway). 

The value is computed at the insertion, as offset is too. Possibly the misleading thing is to have offset as an i-time overload.

I don't think we should have earlysmps as i-time, that gives the wrong idea.

Reply:
i-time offsetsmps certainly gives me the impression both should be available. What I'm saying here is if the first cycle *IS* the last cycle, if the duration of the event is shorter than one ksmps buffer length.

Reply:
In that case, at perf-time earlysmps will be immediately non-zero, but it does not have a meaning at i-time.

Reply:
Is there a use case for having offsetsmps as init-rate then? Maybe best to get rid of that and get rid of any ambiguity and just force users to use both at perf-time.

Reply:
I agree, just k-time opcodes, removes any confusion.

Reply:
overlad removed.